index.js

```diff
@@ -52,13 +52,20 @@ function fastifyCors (fastify, opts, next) {
         return next(error)
       }
 
-      req.corsPreflightEnabled = resolvedOriginOption
-
       // Disable CORS and preflight if false
       if (resolvedOriginOption === false) {
+        req.corsPreflightEnabled = false
         return next()
       }
 
+      // Falsy values are invalid
+      if (!resolvedOriginOption) {
+        return next(new Error('Invalid CORS origin option'))
```

Comment:
Throwing a HTTP 500 for a CORS issue is not an Internal Server Error, it's a bad request.

Reply:
:thinking: I see your point. I choose 500 because it could occur due to invalid options which isn't necessarily related to the request nor the client (4xx).

Reply:
Indeed my bad.